File: api/src/main/java/moe/caa/multilogin/api/logging/impl/DebugLogger.java
package moe.caa.multilogin.api.logging.impl;

import moe.caa.multilogin.api.logging.IMultiloginLogger;
import moe.caa.multilogin.api.logging.LogLevel;
import org.jetbrains.annotations.NotNull;
import org.jetbrains.annotations.Nullable;

public class DebugLogger implements IMultiloginLogger {
    private final IMultiloginLogger innerLogger;

    public DebugLogger(IMultiloginLogger logger) {
        this.innerLogger = logger;
    }

    @Override
    public void log(@NotNull LogLevel level, @Nullable String message, @Nullable Throwable throwable) {
        if (level == LogLevel.DEBUG) {
            innerLogger.log(LogLevel.INFO, "[DEBUG] " + message, throwable);
        } else {
            innerLogger.log(level, message, throwable);
        }
    }
}

File: api/src/main/kotlin/moe/caa/multilogin/api/logger/bridge/DebugLogger.kt
package moe.caa.multilogin.api.logger.bridge

import moe.caa.multilogin.api.logger.Level
import moe.caa.multilogin.api.logger.Logger
import org.jetbrains.annotations.ApiStatus

@ApiStatus.Internal
class DebugLogger(val originLogger: Logger): Logger {
    override fun log(level: Level, message: String?, throwable: Throwable?) {
        if (level == Level.DEBUG) {
            originLogger.log(Level.INFO, message, throwable)
        } else {
            originLogger.log(level, message, throwable)
        }
    }
}